static/app/views/performance/landing/widgets/widgets/singleFieldAreaWidgetMetrics.tsx

```diff
@@ -0,0 +1,130 @@
+import {Fragment, useMemo} from 'react';
+
+import _EventsRequest from 'sentry/components/charts/eventsRequest';
+import {t} from 'sentry/locale';
+import MetricsRequest from 'sentry/utils/metrics/metricsRequest';
+import {decodeList} from 'sentry/utils/queryString';
+import useApi from 'sentry/utils/useApi';
+import _DurationChart from 'sentry/views/performance/charts/chart';
+
+import {GenericPerformanceWidget} from '../components/performanceWidget';
+import {transformMetricsToArea} from '../transforms/transformMetricsToArea';
+import {PerformanceWidgetProps, QueryDefinition, WidgetDataResult} from '../types';
+import {ChartDefinition, PerformanceWidgetSetting} from '../widgetDefinitions';
+
+import {DurationChart, HighlightNumber, Subtitle} from './singleFieldAreaWidget';
+
+type DataType = {
+  chart: WidgetDataResult & ReturnType<typeof transformMetricsToArea>;
+};
+
+export function SingleFieldAreaWidgetMetrics(
+  props: PerformanceWidgetProps & {
+    widgetDefinitions: Record<PerformanceWidgetSetting, ChartDefinition>;
+  }
+) {
+  const api = useApi();
+  const {
+    ContainerActions,
+    eventView,
+    organization,
+    chartSetting,
+    chartColor,
+    chartHeight,
+    fields,
+    widgetDefinitions,
+  } = props;
+
+  const globalSelection = eventView.getGlobalSelection();
+
+  if (fields.length !== 1) {
+    throw new Error(`Single field area can only accept a single field (${fields})`);
+  }
+
+  const field = fields[0];
+
+  // TODO(metrics): make this list complete once api is ready
+  const metricsFieldMap = {
```

Comment:
still want to improve this. Glad to get feedback :)

Reply:
Yeah field map might not be ideal but it works, the only other thing you could do is modify the `widgetDefinitions` to have `metricFields` and only replace it with field, but I honestly prefer leaving all the split code inside the widgets, as there are already a couple branches for metrics already. 🤔